File: PolicyPlus/OpenSection.vb
Public Class OpenSection
    Public SelectedSection As AdmxPolicySection
    Private Sub ButtonOK_Click(sender As Object, e As EventArgs) Handles ButtonOK.Click
        ' Report the selected section
        If OptUser.Checked Or OptComputer.Checked Then
            SelectedSection = If(OptUser.Checked, AdmxPolicySection.User, AdmxPolicySection.Machine)
            DialogResult = DialogResult.OK
        End If
    End Sub
    Private Sub OpenSection_KeyDown(sender As Object, e As KeyEventArgs) Handles Me.KeyDown
        If e.KeyCode = Keys.Escape Then DialogResult = DialogResult.Cancel
    End Sub
    Private Sub OpenSection_Shown(sender As Object, e As EventArgs) Handles Me.Shown
        OptUser.Checked = False
        OptComputer.Checked = False
    End Sub
    Public Function PresentDialog(UserEnabled As Boolean, CompEnabled As Boolean) As DialogResult
        OptUser.Enabled = UserEnabled
        OptComputer.Enabled = CompEnabled
        Return ShowDialog()
    End Function
End Class

File: PolicyPlus/OpenSection.cs
using System;
using System.Windows.Forms;

namespace PolicyPlus
{
    public partial class OpenSection
    {
        public AdmxPolicySection SelectedSection;

        public OpenSection()
        {
            InitializeComponent();
        }
        private void ButtonOK_Click(object sender, EventArgs e)
        {
            // Report the selected section
            if (OptUser.Checked | OptComputer.Checked)
            {
                SelectedSection = OptUser.Checked ? AdmxPolicySection.User : AdmxPolicySection.Machine;
                DialogResult = DialogResult.OK;
            }
        }
        private void OpenSection_KeyDown(object sender, KeyEventArgs e)
        {
            if (e.KeyCode == Keys.Escape)
                DialogResult = DialogResult.Cancel;
        }
        private void OpenSection_Shown(object sender, EventArgs e)
        {
            OptUser.Checked = false;
            OptComputer.Checked = false;
        }
        public DialogResult PresentDialog(bool UserEnabled, bool CompEnabled)
        {
            OptUser.Enabled = UserEnabled;
            OptComputer.Enabled = CompEnabled;
            return ShowDialog();
        }
    }
}